egress/src/reports/idf_event.rs

```diff
@@ -0,0 +1,335 @@
+use std::collections::HashSet;
+
+use axum::{
+    extract::{Path, Query, State},
+    http::StatusCode,
+    Json,
+};
+use chrono::{DateTime, TimeDelta, Utc};
+use futures::{stream::FuturesOrdered, StreamExt};
+use serde::{Deserialize, Serialize};
+use util::{DbPools, PooledPgConn};
+
+use crate::{
+    error::{self, internal_error, not_found_error, Error},
+    patchwork::{self, Patch, PatchworkLabel, PatchworkTimeseriesTables},
+};
+
+use super::idf_station::IdfUnit;
+
+// sum(precipitation_amount PT1M)
+const PRECIPITATION_PARAM_ID: i32 = 105;
+
+// TODO: make sure these defaults are correct
+const DEFAULT_LEVEL: Option<i32> = Some(200);
+const DEFAULT_SENSOR: Option<i32> = Some(0);
```

Comment:
@ketilt Can you double check that these are the parameters used in ODA to calculate the IDF event? I don't remember seeing them explicitly referenced in the old code

Reply:
200 is default level for air temp params, but 1000 would be default for wind stuff... (off the top of my head). Sensor 0 is definetely the default (most recomended).

Reply:
Level is kind of irrelevant for precip though, right?

Reply:
Good point... yeah, I don't think its relevant

Reply:
@pmlefeuvre-met  is the authority on this, but IIRC precipitation does not have a significant relationship with level. Precipitation at ground level should be functionally indistinguishable from, say, 10m up